documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,33 @@ You may obtain a copy of the License at
 
-This works by having each process keep a copy of the data. All updates are written through the StateSynchronizer which appends them to a Pravega segment. They can keep up to date of changes to the data by consuming from the segment. To provide consistency a conditional append is used. This ensures that the updates can only proceed if the process performing them has the most recent data. Finally to prevent the segment from growing without bound, we use have a compact operation that re-writes the latest data, and truncates the old data so it can be dropped.
+In Pravega, the State Synchronizer is used to synchronize data with strong consistency guarantees across machines. The [State Synchronizer API](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java) can be used across various processes to perform updates on that Object. The State Synchronizer ensures that every process that is performing an update on the latest version of that Object. Thus the Object is coordinated across a fleet and everyone sees the same sequence of updates on the same object.
 
-This model works well when most updates are small in comparison to the total data size being stored, as they can be written a small deltas. As with any optimistic concurrency system it would work worst when many processes are all contending to try to update the same information at the same time.
+The idea is to use a Stream to persist a sequence of changes for a shared state. And allow various applications to use the [Pravega Java Client Library](http://pravega.io/docs/latest/state-synchronizer/#shared-state-and-pravega) to concurrently read and write the shared state in a consistent fashion.
+
+This works by having each process keep a copy of the data. All the updates are written through the State Synchronizer which appends them to the Pravega Stream Segment. Latest updates can be incorporated to the data by consuming from the Stream Segment. To provide consistency a conditional append is used. This ensures that the updates can only proceed if the process performing them has the most recent data. To avoid the unbounded data in the Stream Segment, a compact operation is involved which re-writes the latest data and truncates the old data.
+
+In the case of State Synchronizer, it uses the fact that only a single Pravega server can update a Segment at any given time. So it uses the atomic compare-and-set operation on the Segments to build up a higher level abstraction while maintaining the consistency for the larger application.
```

Comment:
"In Pravega, a segment of a stream is always owned by a single server. This allows it to provide  atomic compare-and-set operation on the Segments. This primitive is used to build a higher level abstraction at the application layer while maintaining strong consistency."

Reply:
@tkaitchuck  Accepted suggestion and content edited.Please check